hicommonwealth/commonwealth
# Add search pagination + other fixes

## Link to Issue
Closes: #3449
Closes: #3404
Closes: #3406

## Description of Changes
- Adds pagination for search pages (threads, replies)
- Fixed malformed URI sort bug
- Fixes issue with comment search results not showing full search text

## Test Plan
- From the home page (unscoped), search for a popular term and submit to see the search results page
- Scroll to the bottom and confirm that new results are loaded when bottom is reached (infinite scroll)
- Switch tabs, confirm that tab is updated and URL is updated with tab name
- Navigate to a tabbed URL (http://localhost:8080/search?q=amazing&chainScope=all_chains&sort=Best&tab=Replies) – confirm that the page initializes on the Replies tab
- Repeat steps from beginning, for a specific scope (e.g. dydx, Ethereum)

## Deployment Plan
N/A

## Other Considerations
N/A
